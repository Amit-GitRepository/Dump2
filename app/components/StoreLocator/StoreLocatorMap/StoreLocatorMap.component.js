import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from '../StoreLocatorMap/StoreLocatorMap.style';
import trueShopMapViewTestIDs from '../../../config/testid/trueShopMapViewPage';
import {Button, IconTextInput, InfoBar, List, Map, Modal} from '../../Shared';
import {translate} from '../../../language/i18n/helper';
import {View} from 'react-native';

class StoreLocatorMap extends Component {
  addStoreMarker = (store) => () => this.props.addStoreMarker(store)

  listFilteredStores = (filteredStores) => {
    if (result(filteredStores, '[0].storeName') === translate('STORE_LOCATOR_STORE_NOT_FOUND')) {
      return (
        <View style={styles.listContainer}>
          <List data={filteredStores} headingKey='storeName'/>
        </View>
      );
    } else {
      const storeList = filteredStores.map((store) => {
        store.storeName = store.name[this.props.language] || store.name['en'];
        return store;
      });
      return (
        <View style={styles.listContainer}>
          <List onPress={this.addStoreMarker} data={storeList} headingKey='storeName'/>
        </View>
      );
    }
  }

  render () {
    const {initialRegion, searchModalVisibility, onSearchTextChange, markerList, currentLocation,
      onPressCurrLocation, onSearchPress, filteredStores, hideSearchModal, onCalloutPress, radiusKM, language} = this.props;
    const infoBarText = translate('STORE_LOCATOR__COVERAGE', {distance: radiusKM});
    return (
      <View style={styles.container}>
        <InfoBar text={infoBarText} style={styles.infoBar} accessibilityLabel={trueShopMapViewTestIDs.LABEL__INFOBAR_COVERAGE} testID={trueShopMapViewTestIDs.LABEL__INFOBAR_COVERAGE}/>
        <Map onSearchPress={onSearchPress} initialRegion={initialRegion} markerList={markerList}
          currentLocation={currentLocation} onPressCurrLocation={onPressCurrLocation} onCalloutPress={onCalloutPress} language={language}/>
        <Modal animationIn='fadeInDown' animationOut='fadeOutUp' style={styles.modal} onBackButtonPress={hideSearchModal} isVisible={searchModalVisibility} >
          <View style={styles.modalContent}>
            <View style={styles.searchBar}>
              <View style={styles.searchInput}>
                <IconTextInput containerStyle={styles.inputBar} style={styles.inputText}
                  placeholder={translate('STORE_LOCATOR__SEARCH')} onChangeText={onSearchTextChange}
                  iconName={'search'} iconSize={20} autoFocus/>
              </View>
              <View style={styles.buttonWrapper}>
                <Button type='inline' text={translate('STORE_LOCATOR__CANCEL')} onPress={hideSearchModal}/>
              </View>
            </View>
            {filteredStores.length > 0 ?
              this.listFilteredStores(filteredStores) : null
            }
          </View>

        </Modal>
      </View>
    );
  }
}

StoreLocatorMap.defaultProps = {
  initialRegion: null,
  markerList: [],
  filteredStores: [],
  currentLocation: null,
  addStoreMarker: noop,
  hideSearchModal: noop,
  onPressCurrLocation: noop,
  onSearchTextChange: noop,
  onCalloutPress: noop,
  onSearchPress: noop
};

StoreLocatorMap.propTypes = {
  initialRegion: PropTypes.object,
  markerList: PropTypes.array,
  filteredStores: PropTypes.array,
  currentLocation: PropTypes.object,
  onPressCurrLocation: PropTypes.func,
  hideSearchModal: PropTypes.func,
  onSearchTextChange: PropTypes.func,
  addStoreMarker: PropTypes.func,
  onSearchPress: PropTypes.func,
  onCalloutPress: PropTypes.func,
  searchModalVisibility: PropTypes.bool,
  radiusKM: PropTypes.number,
  language: PropTypes.string
};

export default StoreLocatorMap;
