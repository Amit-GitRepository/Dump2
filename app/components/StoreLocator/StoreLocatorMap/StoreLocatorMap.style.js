import {colors} from '../../../themes/constants.styles';
import {Platform} from '../../../utils/reactNative.util';

export default {
  container: {
    flex: 1,
    marginBottom: -25  // To hide Google logo(in Android) and Legal(in iOS) from Map
  },
  modal: {
    margin: 0,
    paddingTop: Platform.OS === 'ios' ? 20 : 0
  },
  modalContent: {
    alignSelf: 'flex-start',
    flex: 1
  },
  searchBar: {
    flexDirection: 'row',
    alignSelf: 'flex-start',
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    padding: 20
  },
  searchInput: {
    flex: 8
  },
  buttonWrapper: {
    flex: 3,
    alignSelf: 'center',
    marginRight: -5,
    marginLeft: 5
  },
  listContainer: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST
  },
  infoBar: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 1,
    backgroundColor: colors.INFOBAR_WITH_ALPHA
  },
  inputBar: {
    borderRadius: 22,
    paddingLeft: 10,
    borderWidth: 1,
    borderColor: colors.PRIMARY_ACTIONABLE
  },
  inputText: {
    paddingLeft: 5
  }
};
