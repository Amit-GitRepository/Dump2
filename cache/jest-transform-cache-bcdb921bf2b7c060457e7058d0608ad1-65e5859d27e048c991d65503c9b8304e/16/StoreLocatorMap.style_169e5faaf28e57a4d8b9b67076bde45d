Object.defineProperty(exports,"__esModule",{value:true});var cov_89r6kg1tj=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreLocatorMap/StoreLocatorMap.style.js',hash='414751ae32bdea0778dd0fb623b4fa17d7b2efe8',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreLocatorMap/StoreLocatorMap.style.js',statementMap:{},fnMap:{},branchMap:{'0':{loc:{start:{line:11,column:16},end:{line:11,column:46}},type:'cond-expr',locations:[{start:{line:11,column:40},end:{line:11,column:42}},{start:{line:11,column:45},end:{line:11,column:46}}],line:11}},s:{},f:{},b:{'0':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');
var _reactNative=require('../../../utils/reactNative.util');exports.default=

{
container:{
flex:1,
marginBottom:-25},

modal:{
margin:0,
paddingTop:_reactNative.Platform.OS==='ios'?(cov_89r6kg1tj.b[0][0]++,20):(cov_89r6kg1tj.b[0][1]++,0)},

modalContent:{
alignSelf:'flex-start',
flex:1},

searchBar:{
flexDirection:'row',
alignSelf:'flex-start',
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
padding:20},

searchInput:{
flex:8},

buttonWrapper:{
flex:2,
alignSelf:'center',
marginRight:-10,
marginLeft:10},

listContainer:{
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

infoBar:{
position:'absolute',
top:0,
left:0,
right:0,
zIndex:1,
backgroundColor:_constants.colors.INFOBAR_WITH_ALPHA},

inputBar:{
borderRadius:22,
paddingLeft:10,
borderWidth:1,
borderColor:_constants.colors.PRIMARY_ACTIONABLE},

inputText:{
paddingLeft:5}};