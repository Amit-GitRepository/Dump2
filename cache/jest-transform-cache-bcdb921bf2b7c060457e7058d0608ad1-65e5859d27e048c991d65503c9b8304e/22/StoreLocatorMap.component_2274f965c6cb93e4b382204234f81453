Object.defineProperty(exports,"__esModule",{value:true});var cov_biyb85qni=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreLocatorMap/StoreLocatorMap.component.js',hash='9d9c5d57b5f311a999b77fd031a021e417859088',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreLocatorMap/StoreLocatorMap.component.js',statementMap:{'0':{start:{line:12,column:30},end:{line:12,column:68}},'1':{start:{line:12,column:36},end:{line:12,column:68}},'2':{start:{line:15,column:4},end:{line:31,column:5}},'3':{start:{line:16,column:6},end:{line:20,column:8}},'4':{start:{line:22,column:24},end:{line:25,column:8}},'5':{start:{line:23,column:8},end:{line:23,column:78}},'6':{start:{line:24,column:8},end:{line:24,column:21}},'7':{start:{line:26,column:6},end:{line:30,column:8}},'8':{start:{line:36,column:113},end:{line:36,column:123}},'9':{start:{line:37,column:24},end:{line:37,column:82}},'10':{start:{line:38,column:4},end:{line:62,column:6}},'11':{start:{line:66,column:0},end:{line:77,column:2}},'12':{start:{line:79,column:0},end:{line:93,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:19},end:{line:12,column:20}},loc:{start:{line:12,column:30},end:{line:12,column:68}},line:12},'1':{name:'(anonymous_1)',decl:{start:{line:12,column:30},end:{line:12,column:31}},loc:{start:{line:12,column:36},end:{line:12,column:68}},line:12},'2':{name:'(anonymous_2)',decl:{start:{line:14,column:23},end:{line:14,column:24}},loc:{start:{line:14,column:43},end:{line:32,column:3}},line:14},'3':{name:'(anonymous_3)',decl:{start:{line:22,column:43},end:{line:22,column:44}},loc:{start:{line:22,column:54},end:{line:25,column:7}},line:22},'4':{name:'(anonymous_4)',decl:{start:{line:34,column:2},end:{line:34,column:3}},loc:{start:{line:34,column:12},end:{line:63,column:3}},line:34}},branchMap:{'0':{loc:{start:{line:15,column:4},end:{line:31,column:5}},type:'if',locations:[{start:{line:15,column:4},end:{line:31,column:5}},{start:{line:15,column:4},end:{line:31,column:5}}],line:15},'1':{loc:{start:{line:23,column:26},end:{line:23,column:77}},type:'binary-expr',locations:[{start:{line:23,column:26},end:{line:23,column:57}},{start:{line:23,column:61},end:{line:23,column:77}}],line:23},'2':{loc:{start:{line:55,column:13},end:{line:56,column:60}},type:'cond-expr',locations:[{start:{line:56,column:14},end:{line:56,column:53}},{start:{line:56,column:56},end:{line:56,column:60}}],line:55}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0,'11':0,'12':0},f:{'0':0,'1':0,'2':0,'3':0,'4':0},b:{'0':[0,0],'1':[0,0],'2':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _noop=require('lodash/noop');var _noop2=_interopRequireDefault(_noop);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _result=require('lodash/result');var _result2=_interopRequireDefault(_result);
var _StoreLocatorMap=require('../StoreLocatorMap/StoreLocatorMap.style');var _StoreLocatorMap2=_interopRequireDefault(_StoreLocatorMap);
var _trueShopMapViewPage=require('../../../config/testid/trueShopMapViewPage');var _trueShopMapViewPage2=_interopRequireDefault(_trueShopMapViewPage);
var _Shared=require('../../Shared');
var _helper=require('../../../language/i18n/helper');
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

StoreLocatorMap=function(_Component){_inherits(StoreLocatorMap,_Component);function StoreLocatorMap(){var _ref;var _temp,_this,_ret;_classCallCheck(this,StoreLocatorMap);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=StoreLocatorMap.__proto__||Object.getPrototypeOf(StoreLocatorMap)).call.apply(_ref,[this].concat(args))),_this),_this.
addStoreMarker=function(store){cov_biyb85qni.f[0]++;cov_biyb85qni.s[0]++;return function(){cov_biyb85qni.f[1]++;cov_biyb85qni.s[1]++;return _this.props.addStoreMarker(store);};},_this.

listFilteredStores=function(filteredStores){cov_biyb85qni.f[2]++;cov_biyb85qni.s[2]++;
if((0,_result2.default)(filteredStores,'[0].storeName')===(0,_helper.translate)('STORE_LOCATOR_STORE_NOT_FOUND')){cov_biyb85qni.b[0][0]++;cov_biyb85qni.s[3]++;
return(
_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.listContainer},
_react2.default.createElement(_Shared.List,{data:filteredStores,headingKey:'storeName'})));


}else{cov_biyb85qni.b[0][1]++;
var storeList=(cov_biyb85qni.s[4]++,filteredStores.map(function(store){cov_biyb85qni.f[3]++;cov_biyb85qni.s[5]++;
store.storeName=(cov_biyb85qni.b[1][0]++,store.name[_this.props.language])||(cov_biyb85qni.b[1][1]++,store.name['en']);cov_biyb85qni.s[6]++;
return store;
}));cov_biyb85qni.s[7]++;
return(
_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.listContainer},
_react2.default.createElement(_Shared.List,{onPress:_this.addStoreMarker,data:storeList,headingKey:'storeName'})));


}
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(StoreLocatorMap,[{key:'render',value:function render()

{cov_biyb85qni.f[4]++;var _ref2=(cov_biyb85qni.s[8]++,

this.props),initialRegion=_ref2.initialRegion,searchModalVisibility=_ref2.searchModalVisibility,onSearchTextChange=_ref2.onSearchTextChange,markerList=_ref2.markerList,currentLocation=_ref2.currentLocation,onPressCurrLocation=_ref2.onPressCurrLocation,onSearchPress=_ref2.onSearchPress,filteredStores=_ref2.filteredStores,hideSearchModal=_ref2.hideSearchModal,onCalloutPress=_ref2.onCalloutPress,radiusKM=_ref2.radiusKM,language=_ref2.language;
var infoBarText=(cov_biyb85qni.s[9]++,(0,_helper.translate)('STORE_LOCATOR__COVERAGE',{distance:radiusKM}));cov_biyb85qni.s[10]++;
return(
_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.container},
_react2.default.createElement(_Shared.InfoBar,{text:infoBarText,style:_StoreLocatorMap2.default.infoBar,accessibilityLabel:_trueShopMapViewPage2.default.LABEL__INFOBAR_COVERAGE,testID:_trueShopMapViewPage2.default.LABEL__INFOBAR_COVERAGE}),
_react2.default.createElement(_Shared.Map,{onSearchPress:onSearchPress,initialRegion:initialRegion,markerList:markerList,
currentLocation:currentLocation,onPressCurrLocation:onPressCurrLocation,onCalloutPress:onCalloutPress,language:language}),
_react2.default.createElement(_Shared.Modal,{animationIn:'fadeInDown',animationOut:'fadeOutUp',style:_StoreLocatorMap2.default.modal,onBackButtonPress:hideSearchModal,isVisible:searchModalVisibility},
_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.modalContent},
_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.searchBar},
_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.searchInput},
_react2.default.createElement(_Shared.IconTextInput,{containerStyle:_StoreLocatorMap2.default.inputBar,style:_StoreLocatorMap2.default.inputText,
placeholder:(0,_helper.translate)('STORE_LOCATOR__SEARCH'),onChangeText:onSearchTextChange,
iconName:'search',iconSize:20,autoFocus:true})),

_react2.default.createElement(_reactNative.View,{style:_StoreLocatorMap2.default.buttonWrapper},
_react2.default.createElement(_Shared.Button,{type:'inline',text:(0,_helper.translate)('STORE_LOCATOR__CANCEL'),onPress:hideSearchModal}))),


filteredStores.length>0?(cov_biyb85qni.b[2][0]++,
this.listFilteredStores(filteredStores)):(cov_biyb85qni.b[2][1]++,null)))));






}}]);return StoreLocatorMap;}(_react.Component);cov_biyb85qni.s[11]++;


StoreLocatorMap.defaultProps={
initialRegion:null,
markerList:[],
filteredStores:[],
currentLocation:null,
addStoreMarker:_noop2.default,
hideSearchModal:_noop2.default,
onPressCurrLocation:_noop2.default,
onSearchTextChange:_noop2.default,
onCalloutPress:_noop2.default,
onSearchPress:_noop2.default};cov_biyb85qni.s[12]++;


StoreLocatorMap.propTypes={
initialRegion:_propTypes2.default.object,
markerList:_propTypes2.default.array,
filteredStores:_propTypes2.default.array,
currentLocation:_propTypes2.default.object,
onPressCurrLocation:_propTypes2.default.func,
hideSearchModal:_propTypes2.default.func,
onSearchTextChange:_propTypes2.default.func,
addStoreMarker:_propTypes2.default.func,
onSearchPress:_propTypes2.default.func,
onCalloutPress:_propTypes2.default.func,
searchModalVisibility:_propTypes2.default.bool,
radiusKM:_propTypes2.default.number,
language:_propTypes2.default.string};exports.default=


StoreLocatorMap;