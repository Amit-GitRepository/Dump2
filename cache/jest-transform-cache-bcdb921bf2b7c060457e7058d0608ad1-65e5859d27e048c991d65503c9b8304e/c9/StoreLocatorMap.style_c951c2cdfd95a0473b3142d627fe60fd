Object.defineProperty(exports,"__esModule",{value:true});var cov_89r6kg1tj=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreLocatorMap/StoreLocatorMap.style.js',hash='28e6e2c688b2a471e7740157db553a05e7f20c87',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/StoreLocator/StoreLocatorMap/StoreLocatorMap.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
container:{
flex:1,
marginBottom:-25},

modal:{
margin:0},

modalContent:{
alignSelf:'flex-start',
flex:1},

searchBar:{
flexDirection:'row',
alignSelf:'flex-start',
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,
padding:20},

searchInput:{
flex:8},

buttonWrapper:{
flex:2,
alignSelf:'center',
marginRight:-10,
marginLeft:10},

listContainer:{
backgroundColor:_constants.colors.PRIMARY_BG_TEXT_CONTRAST},

infoBar:{
position:'absolute',
top:0,
left:0,
right:0,
zIndex:1,
backgroundColor:_constants.colors.INFOBAR_WITH_ALPHA},

inputBar:{
borderRadius:22,
borderWidth:1,
borderColor:_constants.colors.PRIMARY_ACTIONABLE}};