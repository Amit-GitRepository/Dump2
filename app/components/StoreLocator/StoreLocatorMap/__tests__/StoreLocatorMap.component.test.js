import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import StoreLocatorMap from '../StoreLocatorMap.component';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('StoreLocatorMap component', () => {
  const initialRegion = {
    latitude: 12.9142059,
    longitude: 77.6820747,
    latitudeDelta: 0.012,
    longitudeDelta: 0.034
  };
  it('should render correctly', () => {
    const componentTree = renderer.create(<StoreLocatorMap initialRegion={initialRegion}/>).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
  it('addStoreMarker: should curry props.addStoreMarker', () => {
    const wrapper = shallow(<StoreLocatorMap />);
    const props = {addStoreMarker: jest.fn()};
    wrapper.setProps(props);
    wrapper.instance().addStoreMarker({name: 'someStoreName'})();
    expect(props.addStoreMarker).toBeCalledWith({name: 'someStoreName'});
  });
});
