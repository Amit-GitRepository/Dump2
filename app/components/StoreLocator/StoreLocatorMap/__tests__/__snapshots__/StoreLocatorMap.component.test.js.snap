// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreLocatorMap component should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginBottom": -25,
    }
  }
>
  <View
    accessibilityLabel="trueShopMapView_labelInfobar_coverage"
    style={
      Array [
        Object {
          "backgroundColor": "rgba(40, 60, 70, 0.8)",
        },
        Object {
          "backgroundColor": "rgba(40, 60, 70, 0.8)",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        },
      ]
    }
    testID="trueShopMapView_labelInfobar_coverage"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFFFFF",
          "fontFamily": "SukhumvitSet-Text",
          "fontSize": 18,
          "padding": 2,
          "textAlign": "center",
        }
      }
    >
      Coverage [missing {{distance}} value] k.m.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "bottom": 35,
          "position": "absolute",
          "right": 5,
        }
      }
    >
      <View
        elevation={4}
        style={
          Object {
            "backgroundColor": "#F93E60",
            "borderColor": "#FFFFFF",
            "borderRadius": 100,
            "borderWidth": 2,
            "margin": 5,
            "shadowColor": "#303C46",
            "shadowOffset": Object {
              "height": 2,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessibilityLabel="trueShopMapView_icon_searchStore"
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "borderRadius": 55,
                    "color": "#FFFFFF",
                    "fontSize": 55,
                    "overflow": "hidden",
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              testID="trueShopMapView_icon_searchStore"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(48,60,70,0.9)",
            "height": 1334,
            "width": 750,
          },
        ]
      }
      testID={undefined}
    />
    <View
      onModalHide={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "margin": 0,
            "paddingTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#FFFFFF",
              "flexDirection": "row",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 8,
              }
            }
          >
            <View
              accessibilityLabel={undefined}
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#605E71",
                    "borderRadius": 4,
                    "borderWidth": 2,
                    "flexDirection": "row",
                  },
                  Object {
                    "borderColor": "#F93E60",
                    "borderRadius": 22,
                    "borderWidth": 1,
                    "paddingLeft": 10,
                  },
                ]
              }
              testID={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 20,
                    },
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                      "color": "#B9C1C9",
                      "padding": 5,
                    },
                    Object {
                      "fontFamily": "iservice",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <TextInput
                allowFontScaling={true}
                autoFocus={true}
                onChangeText={[Function]}
                placeholder="Search"
                placeholderTextColor="#B9C1C9"
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "borderRadius": 4,
                      "flex": 1,
                      "fontSize": 16,
                      "paddingHorizontal": 0,
                      "paddingVertical": 10,
                    },
                    Object {
                      "paddingLeft": 5,
                    },
                  ]
                }
                underlineColorAndroid="transparent"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "flex": 3,
                "marginLeft": 5,
                "marginRight": -5,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "borderRadius": 50,
                  "borderWidth": 0,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#F93E60",
                            "fontSize": 16,
                          },
                          Object {},
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
